pkg/roachprod/vm/azure/azure.go

```diff
@@ -1690,3 +1698,25 @@ func MachineFamilyVersionFromMachineType(machineType string) int {
 	}
 	return -1
 }
+
+// MachineSupportsNVMe Azure supports Nvme for E series v5 machine family.
+// OS disk and network disk support nvme. Local storage do not support nvme.
+func MachineSupportsNVMe(machineType string) bool {
+	version := MachineFamilyVersionFromMachineType(machineType)
+	if version == 5 {
+		matches := azureMachineTypes.FindStringSubmatch(machineType)
+		if len(matches) >= 4 {
+			family, features := matches[1], matches[3]
+			// additive features of azure vm are represented by lower case letter
+			// b = Block Storage performance
+			// d = diskful (that is, a local temp disk is present);
+			// s = Premium Storage capable, including possible use of Ultra SSD
+			// https://learn.microsoft.com/en-us/azure/virtual-machines/vm-naming-conventions
+			// example of supported machine types Standard_E2bds_v5, Standard_E2bs_v5
+			if family == "Standard_E" && (features == "bs" || features == "bds") {
```

Comment:
Out of curiosity, was there a particular reason to not use `b` VMs in roachtest? This will always be `false` with our current machine type selection logic.

https://github.com/cockroachdb/cockroach/blob/832ba884372be378e811a20d66f0d39b3d27a3ab/pkg/cmd/roachtest/spec/machine_type.go#L267-L282

cc @srosenberg

Reply:
> Out of curiosity, was there a particular reason to not use b VMs in roachtest? 

I may have missed their availability. If I understand correctly (insofar as it's possible, given Azure's labyrinth of instance types), we could only use `Ebsv5` [1] since it's the only family which supports Cascade Lake and NVMe with remote storage.

[1] https://learn.microsoft.com/en-us/azure/virtual-machines/ebdsv5-ebsv5-series#ebsv5-series-nvme

Reply:
Should i raise a follow up PR to include `Ebsv5` in machine type selection logic?

Reply:
> Should i raise a follow up PR to include `Ebsv5` in machine type selection logic?

Sure, let's create a new tracking issue. Tbh, I'm not too enthusiastic about this instance type. It's nearly double the price of its GCE analog `n2-highmem`, which seems like a high price to pay just to get NVMe. Further, the limited availability means we could only rely on benchmarking "highmem" configurations.

Reply:
Yeah, it seems like it's not worth it for now. I think a `TODO` to revisit this in the `machine_type.go` file is sufficient.